Basic-Car-Maintenance/Shared/Dashboard/ViewModels/DashboardViewModel.swift

```diff
@@ -78,6 +78,26 @@ class DashboardViewModel: ObservableObject {
         }
     }
     
+    func updateEvent(_ maintenanceEvent: MaintenanceEvent) async {
+        
+        if let uid = authenticationViewModel.user?.uid {
+            guard let id = maintenanceEvent.id else { return }
+            var eventToUpdate = maintenanceEvent
+            eventToUpdate.userID = uid
+            do {
+                try Firestore
+                    .firestore()
+                    .collection("maintenance_events")
+                    .document(id)
+                    .setData(from: eventToUpdate)
+            } catch {
+                showAddErrorAlert.toggle()
+                errorMessage = error.localizedDescription
+            }
+        }
+        await self.getMaintenanceEvents()
```

Comment:
This could actually be moved into the do catch, because if there was an error, there's no need to fetch new data, because nothing would be new